src/notsotuf/utils/file_handler.py

```diff
@@ -39,3 +52,26 @@ def start_script_and_exit(
     subprocess.Popen([script_path], creationflags=subprocess.CREATE_NEW_CONSOLE)
     logger.debug('exiting')
     sys.exit(0)
+
+def _start_script_and_exit_mac(
+        src_dir: Union[pathlib.Path, str], dst_dir: Union[pathlib.Path, str]
+):
+    logger.debug(f"Moving content of {src_dir} to {dst_dir}.")
+    for new_file_name in os.listdir(src_dir):
+        _remove_any(os.path.join(dst_dir, new_file_name))
+    shutil.copytree(src_dir, dst_dir, dirs_exist_ok=True)
+    _remove_any(src_dir)
+    logger.debug(f"Restarting application, running {sys.executable}.")
+    subprocess.Popen(sys.executable, shell=True) # nosec
+    sys.exit(0)
+
+def _remove_any(file_or_dir):
```

Comment:
I copied a `remove_path` implementation [from my pyupdater fork][1] into `utils/__init__.py`, including test.

So, you could use: `from notsotuf.utils import remove_path`

However, I'm not sure we should remove the `src_dir` here: The `src_dir` in this case is the `extract_dir`. By default, `extract_dir` is a temporary dir which is removed automatically. The user-specified `src_dir` was actually intended for the case where the user may want to keep the extracted files.

[1]: https://github.com/dennisvang/PyUpdater/blob/a6a54ef3c7919ee5c3c1a2bdf605c967b849d05b/pyupdater/core/utils.py#L64